File: 2024/go/puzzles/dec02.go
package puzzles

import (
	"aoc/utils"
	"fmt"
)

var Dec02 = Puzzle{
	SolvePart1: func(date string, useTestData bool) {
		fmt.Println("Day 02, Part 1")

		lines, err := utils.DataFileReader(date, useTestData, "1", false)
		if err != nil {
			fmt.Println("Error reading file:", err)
			return
		}

		var total int = 0

		var outputString = "Total [using puzzle data]: "

		if useTestData {
			outputString = "Total [using test data]: "
		}

		fmt.Println(outputString, total, "\n\n")

	}, // SolvePart1

	SolvePart2: func(date string, useTestData bool) {
		fmt.Println("Day 02, Part 2")

		lines, err := utils.DataFileReader(date, useTestData, "1", false)
		if err != nil {
			fmt.Println("Error reading file:", err)
			return
		}

		// count the total
		var total int = 0

		var outputString = "Total [using puzzle data]: "

		if useTestData {
			outputString = "Total [using test data]: "
		}

		fmt.Println(outputString, total, "\n\n")

	}, // SolvePart2
}

File: 2024/rust/src/puzzles/dec02.rs
use crate::utils::data_file_reader::data_file_reader;

pub fn solvePart1(date: &str, use_test_data: bool) {
	println!("day 02, Part 1: ");

	// create a data file reader and read the file.
	match data_file_reader(date, use_test_data, 1, false) {
		Ok(lines) => {
			for line in lines {
				
			}
			
			// count the total
			let mut total: i32 = 0;

			let mut outputString: String = String::new();
			if (use_test_data) {
				outputString.push_str("Total [using test data]");
			} else {
				outputString.push_str("Total [using puzzle data]");
			}
			println!("{}: {}", outputString, total);

		}
		Err(_) => {
			println!("Error reading file");
		}
	}
}

pub fn solvePart2(date: &str, use_test_data: bool) {
	println!("day 02, Part 2: ");

	// create a data file reader and read the file.
	match data_file_reader(date, use_test_data, 1, false) {
		Ok(lines) => {
			for line in lines {
				
			}

			// count the total
			let mut total: i32 = 0;

			let mut outputString: String = String::new();
			if (use_test_data) {
				outputString.push_str("Total [using test data]");
			} else {
				outputString.push_str("Total [using puzzle data]");
			}
			println!("{}: {}", outputString, total);

		}
		Err(_) => {
			println!("Error reading file");
		}
	}
}
